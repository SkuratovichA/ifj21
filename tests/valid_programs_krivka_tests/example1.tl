-- Program 1: Vypocet faktorialu (iterativne)
require "ifj21"

function main() -- uzivatelska funkce bez parametru
  local a : integer
  local vysl : integer = 0
  write("Zadejte cislo pro vypocet faktorialu\n")
  a = readi()
  if a == nil then
      write("a je nil\n") return
  else
  end
  if a < 0 then
    write("Faktorial nelze spocitat\n")
  else
    vysl = 1
    while a > 0 do
      vysl = vysl * a a = a - 1  -- dva prikazy
    end
    write("Vysledek je: ", vysl, "\n")
  end
end

main()  -- prikaz hlavniho tela programu