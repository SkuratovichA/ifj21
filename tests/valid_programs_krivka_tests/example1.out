Zadejte cislo pro vypocet faktorialu
Vysledek je: 120
